.github/workflows/dependabot-auto-merge.yml

```diff
@@ -0,0 +1,37 @@
+name: Dependabot auto-merge
+on:
+  workflow_call:
+    inputs:
+      merge-option:
+        description: 'Merge option'
+        type: string
+        required: false
+        default: '--squash'
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.user.login == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.3.3
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Enable auto-merge for minor and patch Dependabot PRs
+        if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-minor' || steps.metadata.outputs.update-type == 'version-update:semver-patch' }}
+        run: gh pr merge --auto ${{ inputs.merge-option }} "$PR_URL"
+        env:
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          # this token CANNOT be the default `GITHUB_TOKEN`, otherwise the merge of the PR will not trigger a build
+          GITHUB_TOKEN: ${{ secrets.DEPENDABOT_GITHUB_TOKEN }}
+      - name: Auto-approve patch Dependabot PRs
+        if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-patch' }}
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          GITHUB_TOKEN: ${{ secrets.DEPENDABOT_GITHUB_TOKEN }}
```

Comment:
@erdemedeiros FYI I made the change here to not use GITHUB_TOKEN (as it will differ from the creator of the PR here anyway)

Reply:
secrets needs to be declared in inputs in a reusable workflow (you don't get any error until you try use it - it's always empty even if you are passing on the call)

https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow

Reply:
oh it's probably working because of `secrets: inherit` but I would suggest to add them anyway otherwise can be confusing if not using `secrets: inherit` on the caller

Reply:
I'm fine following this good practice :+1: 
Now this is not followed by any of the other reusable workflows (also relying on `secrets: inherit` I guess): should this be changed everywhere?
Also I'm not sure how this will behave regarding selection of dependabot secrets (instead of regular secrets): I'd need to test again if that works alright.